tests/unit/sagemaker/cli/compatibility/v2/modifiers/test_training_input.py

```diff
@@ -0,0 +1,113 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+from __future__ import absolute_import
+
+import pasta
+import pytest
+
+from sagemaker.cli.compatibility.v2.modifiers import training_input
+from tests.unit.sagemaker.cli.compatibility.v2.modifiers.ast_converter import ast_call, ast_import
+
+
+@pytest.fixture
+def constructors():
+    return (
+        "sagemaker.session.s3_input(s3_data='s3://a')",
+        "sagemaker.inputs.s3_input(s3_data='s3://a')",
+        "sagemaker.s3_input(s3_data='s3://a')",
+        "session.s3_input(s3_data='s3://a')",
+        "inputs.s3_input(s3_data='s3://a')",
+        "s3_input(s3_data='s3://a')",
+    )
+
+
+@pytest.fixture
+def import_statements():
+    return (
+        "from sagemaker.session import s3_input",
+        "from sagemaker.inputs import s3_input",
+        "from sagemaker import s3_input",
+    )
+
+
+def test_constructor_node_should_be_modified(constructors):
+    modifier = training_input.TrainingInputConstructorRefactor()
+    for constructor in constructors:
+        node = ast_call(constructor)
+        assert modifier.node_should_be_modified(node)
+
+
+def test_constructor_node_should_be_modified_random_call():
+    modifier = training_input.TrainingInputConstructorRefactor()
+    node = ast_call("FileSystemInput()")
+    assert not modifier.node_should_be_modified(node)
+
+
+def test_constructor_modify_node():
+    modifier = training_input.TrainingInputConstructorRefactor()
+
+    node = ast_call("s3_input(s3_data='s3://a')")
+    modifier.modify_node(node)
+    assert "TrainingInput(s3_data='s3://a')" == pasta.dump(node)
+
+    node = ast_call("sagemaker.s3_input(s3_data='s3://a')")
+    modifier.modify_node(node)
+    assert "sagemaker.TrainingInput(s3_data='s3://a')" == pasta.dump(node)
+
+    node = ast_call("session.s3_input(s3_data='s3://a')")
+    modifier.modify_node(node)
+    assert "inputs.TrainingInput(s3_data='s3://a')" == pasta.dump(node)
+
+    node = ast_call("inputs.s3_input(s3_data='s3://a')")
+    modifier.modify_node(node)
+    assert "inputs.TrainingInput(s3_data='s3://a')" == pasta.dump(node)
+
+    node = ast_call("sagemaker.inputs.s3_input(s3_data='s3://a')")
+    modifier.modify_node(node)
+    assert "sagemaker.inputs.TrainingInput(s3_data='s3://a')" == pasta.dump(node)
+
+    node = ast_call("sagemaker.session.s3_input(s3_data='s3://a')")
+    modifier.modify_node(node)
+    assert "sagemaker.inputs.TrainingInput(s3_data='s3://a')" == pasta.dump(node)
```

Comment:
optional - you could also make an iterable of tuples to reduce how much code is repeated
```suggestion
    calls = (
        ("s3_input(s3_data='s3://a')", TrainingInput(s3_data='s3://a')"),
        ("sagemaker.s3_input(s3_data='s3://a')", "sagemaker.TrainingInput(s3_data='s3://a')"),
        ("session.s3_input(s3_data='s3://a')", "inputs.TrainingInput(s3_data='s3://a')"),
        ("inputs.s3_input(s3_data='s3://a')", "inputs.TrainingInput(s3_data='s3://a')"),
        ("sagemaker.inputs.s3_input(s3_data='s3://a')", "sagemaker.inputs.TrainingInput(s3_data='s3://a')"),
        ("sagemaker.session.s3_input(s3_data='s3://a')", "sagemaker.inputs.TrainingInput(s3_data='s3://a')"),
    )

    for call, expected in calls:
        node = ast_call(call)
        modifier.modify_node(node)
        assert expected == pasta.dump(node)
```